packages/babel-traverse/src/path/lib/virtual-types-validator.ts

```diff
@@ -0,0 +1,213 @@
+import type NodePath from "../index";
+import {
+  isBinding,
+  isBlockScoped as nodeIsBlockScoped,
+  isExportDeclaration,
+  isExpression as nodeIsExpression,
+  isFlow as nodeIsFlow,
+  isForStatement,
+  isForXStatement,
+  isIdentifier,
+  isImportDeclaration,
+  isImportSpecifier,
+  isJSXIdentifier,
+  isJSXMemberExpression,
+  isMemberExpression,
+  isRestElement as nodeIsRestElement,
+  isReferenced as nodeIsReferenced,
+  isScope as nodeIsScope,
+  isStatement as nodeIsStatement,
+  isVar as nodeIsVar,
+  isVariableDeclaration,
+  react,
+} from "@babel/types";
+import type * as t from "@babel/types";
+const { isCompatTag } = react;
+import type { VirtualTypeAliases } from "./virtual-types";
+
```

Comment:
Before this PR I have to mentally remember the link from typing to implementation
> path.isBindingIdentifier => virtualTypes["isBindingIdentifier"].checkType

Now they are placed together. However, the typings and implementations can not be merged because TS does not support `this` assertion in non-class functions, although they will be eventually injected to the `NodePath` prototype.